tensorflow/core/kernels/linalg/matrix_triangular_solve_op_impl.h

```diff
@@ -409,12 +409,32 @@ struct LaunchBatchMatrixTriangularSolve<GPUDevice, Scalar> {
       }
     }
 #elif TENSORFLOW_USE_ROCM
-    for (int batch = 0; batch < batch_size; ++batch) {
+    if (batch_size == 1) {
       OP_REQUIRES_OK(
           context,
           solver->Trsm(side, uplo, trans, diag, colmajor_rows, colmajor_cols,
-                       &alpha, a_ptrs[batch], leading_dim_matrix /*lda*/,
-                       out_ptrs[batch], leading_dim_output /*ldb*/));
+                       &alpha, a_ptrs[0], leading_dim_matrix /*lda*/,
+                       out_ptrs[0], leading_dim_output /*ldb*/));
+    } else {
+      
+      const int kMaxMatrixSizeToBatchSizeRatio = 128;
+      const bool use_batched_solver =
+          m <= kMaxMatrixSizeToBatchSizeRatio * batch_size;
+      if (use_batched_solver) {
+        OP_REQUIRES_OK(
+            context, solver->TrsmBatched(
+                         side, uplo, trans, diag, colmajor_rows, colmajor_cols,
+                         &alpha, &a_ptrs[0], leading_dim_matrix /*lda*/,
+                         &out_ptrs[0], leading_dim_output /*ldb*/, batch_size));
+      } else {
+        for (int batch = 0; batch < batch_size; ++batch) {
+          OP_REQUIRES_OK(
+              context, solver->Trsm(side, uplo, trans, diag, colmajor_rows,
+                                    colmajor_cols, &alpha, a_ptrs[batch],
+                                    leading_dim_matrix /*lda*/, out_ptrs[batch],
+                                    leading_dim_output /*ldb*/));
+        }
+      }
```

Comment:
The code in the TENSORFLOW_USE_ROCM case is exactly the same as in the GOOGLE_CUDA case. 

So I think you can simply get rid of the TENSORFLOW_USE_ROCM case entirely (also remove the #GOOGLE_CUDA directive on line 376).